lib/http/features/auto_inflate.rb

```diff
@@ -17,15 +17,15 @@ def wrap_response(response)
           :headers       => response.headers,
           :proxy_headers => response.proxy_headers,
           :connection    => response.connection,
-          :body          => stream_for(response.connection),
+          :body          => stream_for(response.connection, :encoding => response.body.encoding),
```

Comment:
Let's move logic to `stream_for`:

```suggestion
          :body          => wrap_body(response),
```

Then move and rename `stream_for` to be `private` `wrap_body`:

``` ruby
private

def wrap_body(response)
  stream = Response::Inflater.new(response.connection)

  Response::Body.new(stream, :encoding => response.body.encoding)
end
```